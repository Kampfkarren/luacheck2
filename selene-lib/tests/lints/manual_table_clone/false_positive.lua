local function falsePositive1(...)
	local result = {}

	for i = 1, select("#", ...) do
		local dictionary = select(i, ...)
		for key, value in pairs(dictionary) do
			result[key] = value
		end
	end

	return result
end

local function falsePositive2(t)
	local result = {}
	local count = 0

	while count < 20 do
		count = count + 1
		for key, value in pairs(t) do
			result[key] = value
		end
	end

	return result
end

local function falsePositive3(t)
	local result = {}
	local count = 0

	repeat
		count = count + 1
		for key, value in pairs(t) do
			result[key] = value
		end
	until count > 20

	return result
end

local result4 = {}
local function falsePositive4(t)
    for key, value in t do
        result3[key] = value
    end
end

local result5 = {}
local function falsePositive5(t)
	local function f() end

    for key, value in t do
        result4[key] = value
    end
end

local function falsePositive6(t)
    local result = {}

	if b then return end

    if a then
        for key, value in t do
            result4[key] = value
        end
    end
end

local function notFalsePositive1(t)
	local result = {}

	for i = 1, 10 do
		print(i)
	end

	for key, value in pairs(t) do
		result[key] = value
	end

	return result
end

local function notFalsePositive2(t)
	for i = 1, 10 do
		local result = {}

		for key, value in pairs(t) do
			result[key] = value
		end
	end
end
