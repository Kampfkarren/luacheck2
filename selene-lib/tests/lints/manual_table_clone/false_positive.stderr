error[manual_table_clone]: manual implementation of table.clone
   ┌─ false_positive.lua:77:2
   │  
71 │       local result = {}
   │       ----------------- remove this definition
   ·  
77 │ ╭     for key, value in pairs(t) do
78 │ │         result[key] = value
79 │ │     end
   │ ╰───────^
   │  
   = try `local result = table.clone(t)`

error[manual_table_clone]: manual implementation of table.clone
   ┌─ false_positive.lua:86:3
   │  
86 │ ╭         local result = {}
87 │ │ 
88 │ │         for key, value in pairs(t) do
89 │ │             result[key] = value
90 │ │         end
   │ ╰───────────^
   │  
   = try `local result = table.clone(t)`

